redfish/provider/resource_redfish_storage_volume.go

```diff
@@ -1,5 +1,757 @@
+
+func updateRedfishStorageVolume(ctx context.Context, service *gofish.Service,
+	d *models.RedfishStorageVolume, state *models.RedfishStorageVolume,
+) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	// Lock the mutex to avoid race conditions with other resources
+	redfishMutexKV.Lock(d.RedfishServer[0].Endpoint.ValueString())
+	defer redfishMutexKV.Unlock(d.RedfishServer[0].Endpoint.ValueString())
+
+	// Get user config
+	storageID := d.StorageControllerID.ValueString()
+	volumeName := d.VolumeName.ValueString()
+	readCachePolicy := d.ReadCachePolicy.ValueString()
+	writeCachePolicy := d.WriteCachePolicy.ValueString()
+	diskCachePolicy := d.DiskCachePolicy.ValueString()
+	applyTime := d.SettingsApplyTime.ValueString()
+
+	var driveNames []string
+	diags.Append(d.Drives.ElementsAs(ctx, &driveNames, true)...)
+
+	volumeJobTimeout := d.ResetTimeout.ValueInt64()
+
+	// Get storage
+	storage, err := getStorage(service, storageID)
+	if err != nil {
+		diags.AddError("Error when retreiving the Systems from the Redfish API", err.Error())
+		return diags
+	}
+
+	// Check if settings_apply_time is doable on this controller
+	err = checkSettingsApplyTime(storage, applyTime)
+	if err != nil {
+		diags.AddError("Error while checking support for settings_apply_time", err.Error())
+		return diags
+	}
+
+	payload := map[string]interface{}{
+		"ReadCachePolicy":  readCachePolicy,
+		"WriteCachePolicy": writeCachePolicy,
+		"DisplayName":      volumeName,
+		"Oem": map[string]map[string]map[string]interface{}{
+			"Dell": {
+				"DellVolume": {
+					"DiskCachePolicy": diskCachePolicy,
+				},
+			},
+		},
+		"Name": volumeName,
+		"@Redfish.SettingsApplyTime": map[string]interface{}{
+			"ApplyTime": applyTime,
+		},
+	}
+
+	// Update volume job
+	jobID, err := updateVolume(service, state.ID.ValueString(), payload)
+	if err != nil {
+		diags.AddError("Error when updating the virtual disk on disk controller", err.Error())
+		return diags
+	}
+
+	// Immediate or OnReset scenarios
+	if applyTime == string(redfishcommon.OnResetApplyTime) { // OnReset case
+		resetType := d.ResetType.ValueString()
+		resetTimeout := d.ResetTimeout.ValueInt64()
+
+		// Reboot the server
+		pOp := powerOperator{ctx, service}
+		_, err := pOp.PowerOperation(resetType, resetTimeout, intervalStorageVolumeJobCheckTime)
+		if err != nil {
+			diags.AddError(RedfishJobErrorMsg, err.Error())
+			return diags
+		}
+	}
+
+	// Wait for the job to finish
+	err = common.WaitForJobToFinish(service, jobID, intervalStorageVolumeJobCheckTime, volumeJobTimeout)
+	if err != nil {
+		diags.AddError(RedfishJobErrorMsg, err.Error())
+		return diags
+	}
+
+	// Get storage volumes
+	volumes, err := storage.Volumes()
+	if err != nil {
+		diags.AddError("there was an issue when retrieving volumes", err.Error())
+		return diags
+	}
+	volumeID, err := getVolumeID(volumes, volumeName)
+	if err != nil {
+		diags.AddError("Error. The volume ID with volume name %s on %s controller was not found", err.Error())
```

Comment:
These %s are not getting populated with info

Reply:
updated.